js/base/Exchange.js

```diff
@@ -2212,6 +2212,177 @@ module.exports = class Exchange {
         });
     }
 
+    safePosition (position, market = undefined) {
+        market = this.market(position['symbol']);
+        const side = position['side'];
+        
+        // essential fields
+        const realizedPnlString = this.safeString (position, 'realizedPnl');
+        const entryPriceString = this.safeString (position, 'entryPrice');
+        const markPriceString = this.safeString (position, 'markPrice');
+        const lastPriceString = this.safeString (position, 'lastPrice');
+        
+        // 'contractSize'
+        let contractSizeString = this.safeString (position, 'contractSize');
+        if (contractSizeString === undefined) {
+            contractSizeString = this.safeString (market, 'contractSize');
+        }
+        // 'contract'
+        const contractsAmountString = this.safeString (position, 'contracts');
+        const contractsMultipliedSize = Precise.stringMul (contractsAmountString, contractSizeString);
+
+        // 'notional'
+        let notionalString = this.safeString (position, 'notional');
+        if (notionalString === undefined) {
+            // TODO CONFLICT: some exchanges only use 'contractsAmount'
+            if (contractsMultipliedSize !== undefined) {
+                const useLastPrice = this.safeValue (this.options, 'useLastPriceForNotional', false);
```

Comment:
I'd suggest that we don't introduce this option and instead do it in the following manner:

- if markPrice is available – use it to calculate the notional
- else if lastPrice is available – fall back to that to calculate the notional
- else leave notional undefined

Reply:
but are we sure that availability of field should decide the calculation, as opposed to pre-determined approach? having that determined in `options` should be any issue for end users (they dont need to know the existence of that option at all, as it's for our internal usage). but anyway, oky, i will do that way.